File: in_Java/0222 Count Complete Tree Nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int countNodes(TreeNode root) {
        if(root == null) return 0;
        int res = 1;
        int rightn = height(root.right);
        int h = height(root);
        int leftn = h-1;
        
        if(leftn == rightn){
            res += Math.pow(2,leftn)-1;
            res += bfs(root.right);
        }else{
            res += Math.pow(2,rightn)-1;
            res += bfs(root.left);
        }
        return res;
    }
    private int height(TreeNode t){
        int res = 0;
        while(t != null){
            res++;
            t = t.left;
        }
        return res;
    }
    private int bfs(TreeNode r){
        if(r == null) return 0;
        Queue<TreeNode> q = new LinkedList<>();
        q.add(r);
        TreeNode cur;
        int n;
        int res = 0;
        while(!q.isEmpty()){
            n = q.size();
            while(n-- != 0){
                cur = q.poll();
                res++;
                if(cur.left != null) q.add(cur.left);
                if(cur.right != null) q.add(cur.right);
            }
        }
        return res;
    }
}

File: in_Python/0222 Count Complete Tree Nodes.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def countNodes(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        if root == None:
            return 0
        left = root
        right = root
        h = 0
        while right:
            left = left.left
            right = right.right
            h += 1
        if left == None:
            return pow(2,h) - 1
        return 1 + self.countNodes(root.left) + self.countNodes(root.right)
        
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def countNodes(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        res = 0
        if root == None:
            return res
        q = [root]
        while q:
            n = len(q)
            while n != 0:
                n -= 1
                h = q.pop(0)
                if h.left:
                    q.append(h.left)
                if h.right:
                    q.append(h.right)
                res += 1
        return res